components/clarinet-sdk-wasm/src/core.rs

```diff
@@ -888,25 +905,33 @@ impl SDK {
     }
 
     #[wasm_bindgen(js_name=runSnippet)]
-    pub fn run_snippet(&mut self, snippet: String) -> String {
+    pub fn run_snippet(&mut self, snippet: String) -> Result<TransactionRes, String> {
```

Comment:
Hmm, this is a breaking change, the only of this PR.

But it's an important change and I certainly don't want to release Clarinet 3.0.

`runSnippet` isn't really used and there is no real use case for that in a unit tests context (what the clarinet-sdk has been used for so far). But it's more needed in a browser context (to build docs, repl, etc). But it's still a breaking change.
Although I don't really like the function name `runSnippet`. Maybe I coud just deprecate it and introduce a new function instead, like `simnet.execute()` (if I do so I have to make sure that the function name _is_ better).

Thoughts @janniks @MicaiahReid?